apps/keyboard/js/keyboard/user_press_manager.js

```diff
@@ -234,11 +256,73 @@ UserPressManager.prototype._handleFinishPress = function(el, coords, id) {
   press.updateCoords(coords,
     press.moved || this._distanceReachesLimit(id, coords));
 
-  if (typeof this.onpressend === 'function') {
-    this.onpressend(press, id);
+  if(this._lowEndDevice && this._exceedSpeedLimit(press) &&
```

Comment:
The other tricky thing here is that `_isLowEndDevice` might be undefined if the user is faster than mozSettings. This can't happen in real world but it happens in tests. The easy way to fix this is to fill a default value on line 79 and use it. The hard way to fix this would be defer the handing of user events into a queue and only process then when we get the setting, but that's too much work.